File: microservices/company/src/main/java/com/illunex/invest/company/controller/PluginControllerImpl.java
package com.illunex.invest.company.controller;

import com.illunex.invest.api.common.response.ResponseData;
import com.illunex.invest.api.common.response.ResponseList;
import com.illunex.invest.api.core.company.controller.PluginController;
import com.illunex.invest.api.core.company.request.PluginRequest;
import com.illunex.invest.company.service.PluginService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
public class PluginControllerImpl implements PluginController {
    private final PluginService pluginService;

    @Override
    public ResponseEntity<ResponseData> savePlugins(PluginRequest request) {
        return ResponseEntity.ok(ResponseData.builder()
                .errorCode(0)
                .message("success")
                .data(pluginService.savePlugin(request))
                .build());
    }

    @Override
    public ResponseEntity<ResponseData> togglePlugins(Long id) {
        pluginService.togglePlugin(id);
        return ResponseEntity.ok(ResponseData.builder()
                .errorCode(0)
                .message("success")
                .build());
    }

    @Override
    public ResponseEntity<ResponseList> getPlugins(Long companyIdx) {
        return ResponseEntity.ok(new ResponseList<>(0
                , "success"
                , pluginService.getPlugins(companyIdx)));
    }
}

File: microservices/shop/src/main/kotlin/com/illunex/invest/shop/controller/PluginControllerImpl.kt
package com.illunex.invest.shop.controller

import com.illunex.invest.api.common.response.ResponseList
import com.illunex.invest.api.core.shop.controller.PluginController
import com.illunex.invest.shop.service.ShopPluginService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.RestController
import java.util.stream.Collectors

@RestController
class PluginControllerImpl: PluginController {
    @Autowired var shopPluginService:ShopPluginService? = null

    override fun getRoleNameByIds(ids: List<String>): ResponseEntity<ResponseList<String>> {
        return ResponseEntity.ok(ResponseList(0
                , "success"
                , shopPluginService!!.getRoleNamesByPluginId(ids.stream()
                    .map { s: String -> s.toLong() }
                    .collect(Collectors.toList()))
        ))
    }
}